tests/test_unit/test_reports.py

```diff
@@ -4,28 +4,136 @@
 
 
 @pytest.mark.parametrize(
-    "data_selection",
+    "valid_dataset",
     [
-        lambda ds: ds.position,  # Entire dataset
-        lambda ds: ds.position.sel(
-            individuals="ind1"
-        ),  # Missing "individuals" dim
-        lambda ds: ds.position.sel(
-            keypoints="key1"
-        ),  # Missing "keypoints" dim
-        lambda ds: ds.position.sel(
-            individuals="ind1", keypoints="key1"
-        ),  # Missing "individuals" and "keypoints" dims
+        "valid_poses_dataset",
+        "valid_bboxes_dataset",
+        "valid_poses_dataset_with_nan",
+        "valid_bboxes_dataset_with_nan",
     ],
 )
-def test_report_nan_values(
-    capsys, valid_poses_dataset_with_nan, data_selection
+@pytest.mark.parametrize(
+    "data_selection, list_expected_individuals_idcs",
+    [
+        (lambda ds: ds.position, [0, 1]),
+        # Report for the full dataset
+        (
+            lambda ds: ds.position.isel(individuals=0),
+            [0],
+        ),  # Report for individual 0 only
+    ],
+)
+def test_report_nan_values_in_position_selecting_individual(
+    capsys,
+    valid_dataset,
+    data_selection,
+    list_expected_individuals_idcs,
+    request,
+):
+    """Test that the nan-value reporting function handles position data
+    with specific ``individuals`` , and that the dataset name (position)
+    and only the relevant individuals are included in the report.
+    """
+    # apply selection to extract relevant position data
+    input_dataset = request.getfixturevalue(valid_dataset)
+    output_data_array = data_selection(input_dataset)
+
+    # produce report
+    report_str = report_nan_values(output_data_array)
+
+    # check report of nan values includes dataset name
```

Comment:
that's the data array name, right? not the dataset name

Reply:
both in the comment and in the assertion error message.

Reply:
yes, that was inherited from the previous test - fixing now